### DISCUSSION TITLE
Orphaned/zombie LocalTaskJob

### DISCUSSION THREAD

**Discussion by Mzhussiyev** [apache/airflow#40211] [General]
Several times I spot zombie/orphaned `LocalTaskJobs` when I was launching an Airflow. There are no active running TI and DAG runs but there are LocalTaskJobs with `running` state (see the screenshot below). 

I thought that such orphaned tasks/jobs should be killed by the scheduler (`_find_zombies` function). 
If you look at job runs their last heartbeat were a while ago and these jobs are still alive with running state, the current date and time you can get from latest heartbeat of `SchedulerJob`. 

So, the questions are:
- why these jobs weren't killed by scheduler
- which process is responsible to look after such orphaned jobs (which probably lost the connection with tasks)
- are these "running" jobs affect to Airflow performance by consuming some resources
- what should we consider in such cases (mark them as 'success' directly in DB?)

![image](https://github.com/apache/airflow/assets/73746305/09723034-2a11-4262-a55d-62491f2f0fda)

**Comment by arollet22** [comment]
I'm experiencing exactly the same issue. Our Airflow is deployed on MWAA
